src/envvars.zig

```diff
@@ -0,0 +1,311 @@
+
+                const env_str = raw_env.?;
+
+                inline for (true_values) |tv| {
+                    if (std.mem.eql(u8, env_str, tv)) {
+                        value.store(.yes, .monotonic);
+                        return true;
+                    }
+                }
+
+                value.store(.no, .monotonic);
+                return false;
+            }
+        },
+    };
+
+    return struct {
+        const Self = @This();
+
+        /// Attempt to retrieve the value of the environment variable for the current platform, if
+        /// the current platform has a supported definition. Returns null otherwise, unlike the
+        /// other methods which will fail at compile time if the platform is unsupported.
+        pub fn platformGet() ReturnType {
+            if (bun.Environment.isPosix) {
+                if (posix_key == null) return null;
+            }
+
+            if (bun.Environment.isWindows) {
+                if (windows_key == null) return null;
+            }
+
+            return get();
+        }
+
+        /// Retrieve the key of the environment variable for the current platform.
+        pub fn key() KeyType {
+            assertPlatformSupported();
+            return Self.platformKey().?;
+        }
+
+        /// Retrieve the key of the environment variable for the current platform, if any.
+        pub fn platformKey() ?KeyType {
+            if (bun.Environment.isPosix) {
+                return posix_key;
+            }
+
+            if (bun.Environment.isWindows) {
+                return windows_key;
+            }
+
+            return null;
+        }
+
+        /// Retrieve the value of the environment variable, loading it if necessary.
+        /// Fails if the current platform is unsupported.
+        pub fn get() ReturnType {
+            assertPlatformSupported();
+
+            switch (Cache.getCached()) {
+                .undefined => return getForceReload(),
+                .not_set => {
+                    if (opts.default != null) return default();
+                    return null;
+                },
+                .value => |v| return v,
+            }
+        }
+
+        /// Retrieve the value of the environment variable, reloading it from the environment.
+        /// Fails if the current platform is unsupported.
+        pub fn getForceReload() ReturnType {
+            assertPlatformSupported();
+            const env_var = bun.getenvZ(key());
+            const maybe_reloaded = Cache.reload(env_var);
+
+            if (maybe_reloaded) |v| return v;
+            if (opts.default != null) return default();
+
+            return null;
+        }
+
+        /// Fetch the default value of this environment variable, if any.
+        ///
+        /// It is safe to compare the result of .get() to the result of .default() to determine if
+        /// the variable is set to its default value, if .default() does not return null.
+        pub fn default() ReturnType {
```

Comment:
This function seems to only be used inside a guard of `opts.default != null`. Maybe it should be removed?

ie

```diff
-                .not_set => {
-                    if (opts.default != null) return default();
-                    return null;
-                },
+               .not_set => return opts.default,
```

or `if (opts.default) |default| return default;`

Reply:
Kind of my reasoning for this is so that a public interface could query for the default value. So users could query a default value, if they so desired